### PR TITLE
Na/content conformance cli enhancement

### PR DESCRIPTION
**Pull request by nayres**
🎟️ [Asana Task](https://app.asana.com/0/1203331966445516/1203575420803095)

---

## Description

This PR changes the CLI interface to match the README

```
$ hc-content [options] [path/to/file] [path/to/file]

Options:
  --cwd
  --config ./path/to/conformance.config.mjs
```

## PR Checklist 🚀

- [ ] Conduct thorough self-review.
- [ ] Add or update tests as appropriate.
- [ ] Write a useful description (above) to give reviewers appropriate context.
- [ ] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

File: packages/content-conformance/src/cli.ts [hunk 1]
```diff
@@ -19,32 +19,27 @@ yargs(hideBin(process.argv)).command(
...
       type: 'string',
     },
-    files: {
-      description: 'Path to content files',
-      default: [],
-      type: 'array',
-    },
   },
   async (argv): Promise<void> => {
+    const [...files] = <string[]>argv._
```
Comment (nayres) [inline packages/content-conformance/src/cli.ts:24]: This allows us to pass rest args. In this case an array of files.

before:
`hc-content content-check --cwd --config --files`

after:
`hc-content --cwd --config [path/to/file]`

Comment (BRKalow) [inline packages/content-conformance/src/cli.ts:24]: Minor: if you are meaning to cast `argv._` here, I think this is more clear:
```suggestion
    const [...files] = argv._ as string[]
```

File: packages/content-conformance/src/cli.ts [hunk 2]
```diff
@@ -8,7 +8,7 @@ import { ContentConformanceRunner } from './runner.js'
  * The CLI interface for our content conformance runner. Should remain a pass-through to ContentConformanceRunner if at all possible to keep the CLI and JS API in-sync
  */
 yargs(hideBin(process.argv)).command(
-  'content-check',
+  ['$0', 'content-check'],
```
Comment (BRKalow) [inline packages/content-conformance/src/cli.ts:11]: 👏

**Review by BRKalow — APPROVED** [review]
👏

**Pull request merged**